hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1245,54 +1178,52 @@ def minimize_individual_blocks(self, i):
             full=False,
         )
 
-    @defines_shrink_pass(
-        lambda self: [
-            (i, self.stable_identifier_for_example(ex))
-            for i in hrange(len(self.blocks))
-            if self.is_shrinking_block(i)
-            for ex in self.examples
-            if ex.length > 0 and ex.start >= self.blocks[i].end
-        ]
-    )
-    def example_deletion_with_block_lowering(self, block_i, ex_id):
-        """Sometimes we get stuck where there is data that we could easily
-        delete, but it changes the number of examples generated, so we have to
-        change that at the same time.
-
-        We handle most of the common cases in try_shrinking_blocks which is
-        pretty good at clearing out large contiguous blocks of dead space,
-        but it fails when there is data that has to stay in particular places
-        in the list.
-
-        This pass exists as an emergency procedure to get us unstuck. For every
-        example and every block not inside that example it tries deleting the
-        example and modifying the block's value by one in either direction.
-        """
-        if block_i >= len(self.blocks):
-            return
-        block = self.blocks[block_i]
-        ex = self.example_for_stable_identifier(ex_id)
-        if ex is None:
+        if self.shrink_target is not initial:
             return
-        u, v = block.bounds
 
-        n = int_from_bytes(self.shrink_target.buffer[u:v])
-        if n == 0:
+        lowered = (
+            self.buffer[: block.start]
+            + int_to_bytes(
+                int_from_bytes(self.buffer[block.start : block.end]) - 1, block.length
+            )
+            + self.buffer[block.end :]
+        )
+        attempt = self.cached_test_function(lowered)
+        if (
+            attempt.status < Status.VALID
+            or len(attempt.buffer) == len(self.buffer)
+            or len(attempt.buffer) == block.end
+        ):
             return
 
-        buf = bytearray(self.shrink_target.buffer)
-        buf[u:v] = int_to_bytes(n - 1, v - u)
+        # If it were then the lexical shrink should have worked and we could
+        # never have got here.
+        assert attempt is not self.shrink_target
+
+        lo = 0
```

Comment:
Consider caching this binary search as well, if you think it's worthwhile.

Reply:
Might as well. Gives me an excuse to make the caching mechanism a bit less shonky.

My suspicion is that it's not actually very helpful, but it can't hurt!